### ISSUE TITLE
Problem using existing DBFVFP index

### ISSUE THREAD

**Issue by cpyrgas** [X-Sharp/XSharpPublic#473]
Try using the attached VFP dbf/cdx files, there's a problem using the index file, see inline comments:

```
FUNCTION Start() AS VOID
LOCAL cDbf AS STRING

RddSetDefault("DBFVFP")

cDbf := "C:\...\temp"
DbUseArea(,,cDbf,,FALSE)

// this causes a System.ArgumentOutOfRangeException: Number was less than the array's lower bound in the first dimension.
// ? DbReindex()

? DbSetOrder(1) // true
? FieldName(1) // SERIES
? DbOrderInfo(DBOI_EXPRESSION) // series, looks good
? FieldGet(1) // 1, ok

// Unhandled Exception: System.IndexOutOfRangeException: Index was outside the bounds of the array.
? DbSeek(1) 

/*
Unhandled Exception: System.IndexOutOfRangeException: Index was outside the bounds of the array.
   at XSharp.RDD.DoubleStruct.SaveToIndex(Byte[] buffer)
   at XSharp.RDD.CDX.CdxTag._ToString(Object toConvert, Int32 sLen, Byte[] buffer, Int32& resultLength)
   at XSharp.RDD.CDX.CdxTag.Seek(DbSeekInfo seekInfo)
   at XSharp.RDD.DBFCDX.Seek(DbSeekInfo seekInfo)
   at XSharp.CoreDb.<>c__DisplayClass94_0.<Seek>b__0()
   at XSharp.CoreDb.Do[T](Func`1 action)
   at XSharp.CoreDb.Seek(Object oValue, Boolean lSoftSeek, Boolean lLast)
   at XSharp.RT.Functions.DbSeek(__Usual[] Xs$Args)
*/
```

[Tempdbf.zip](https://github.com/X-Sharp/XSharpPublic/files/5078545/Tempdbf.zip)

**Comment by RobertvanderHulst** [comment]
It seems that VFP stores the keys for a field of type "I" (Integer) in a different way than I expected.

**Comment by RobertvanderHulst** [comment]
Fixed in e8114ca5e9746c2c582b61c7caaa9395d3a9ecbb
